### DISCUSSION TITLE
Pre-install extension on theia

### DISCUSSION THREAD

**Discussion by wassimkha** [eclipse-theia/theia#10556] [Q&A] [answered]
Hi,

I'm trying to start Theia with some pre-installed extension. I.e we can install VS code extensions inside of theia after it's been compiled:

![Screen Shot 2021-12-16 at 3 20 19 PM](https://user-images.githubusercontent.com/43592500/146443512-77c7c57b-07c7-480e-bf4f-e0163dca280b.png)

Is there a way that when the user opens Theia, we already have some extensions pre-installed for them , how can I do that?

Thanks

**Comment by msujew** [comment, marked as answer]
Hi @wassimkha,

you can simply add extensions you want to have pre-installed to your list of `theiaPlugins` in your `package.json` as shown in the [docs](https://theia-ide.org/docs/composing_applications/#consuming-vs-code-extensions). Call `theia download:plugins` or create a script and call that to actually download them.

**Reply by wassimkha** [reply]
thanks!
